### PR TITLE
Enforce pypoetry

### LINKED ISSUE #2061: Making Poetry mandatory
Hi folks,

ever since I introduced the compatibility fix for macOS 10.13 back when I first joined the team, we've started to rely on external package/environment managers for Python — which eventually ended up including Poetry for virtual environment management.

As of now, we have both a `requirements.txt` and a `pyproject.toml` which _should_  in theory contain the same dependencies, but in practice they go out of sync if one developer forgets about this "constraint". To match our development and testing environments with our build environment, I'm phasing out the `requirements.txt` file and make using Poetry mandatory.

Also see: #2057

### PR DESCRIPTION
**Pull request by archanyhm**
closes #2061

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -2,24 +2,24 @@
 
 [[package]]
 name = "altgraph"
-version = "0.17.3"
+version = "0.17.4"
```
Comment (Lixxis) [inline poetry.lock:5]: version update on that package was intended?

**Review by Lixxis — COMMENTED** [review]
In general, are all the version changes are intended? (safety proof)

File: poetry.lock (hunk 1 above)
Comment (archanyhm) [reply poetry.lock:5]: All updates are according to the specifications in the pyproject.toml; just the lock file got updated and that on purpose, yeah. I think that hasn't happened in a while.

**Comment by larkgz** [comment]
Can you also modify the CONTRIBUTING.md so it’s consistent?

**Comment by archanyhm** [comment]
@larkgz 
> Can you also modify the CONTRIBUTING.md so it’s consistent?

Done!

File: CONTRIBUTING.md
Comment (Thlumyn) [inline CONTRIBUTING.md:1]: Extra comma

**Review by Lixxis — APPROVED** [review]
I don't see any issues, but didn't run the workflows. 
I saw you tested with another developer in discord.

**Pull request merged**